test/test-price-risk-module.js

```diff
@@ -157,196 +180,100 @@ describe("Test PriceRiskModule contract", function () {
   });
 
-  it("Should allow address(0) for the reference oracle", async () => {
-    const { pool, premiumsAccount } = await helpers.loadFixture(deployPoolFixture);
-
-    const PriceRiskModule = await hre.ethers.getContractFactory("PriceRiskModule");
-
-    const rm = await addRiskModule(pool, premiumsAccount, PriceRiskModule, {
-      extraConstructorArgs: [
-        "0x82a6c4AF830caa6c97bb504425f6A66165C2c26e",
-        hre.ethers.constants.AddressZero,
-        _W("0.01"),
-      ],
-      extraArgs: [HOUR],
-    });
-
-    expect(await rm.referenceOracle()).to.equal(hre.ethers.constants.AddressZero);
-  });
-
-  it("Should calculate exchange rate for single asset with no reference", async () => {
-    const { pool, premiumsAccount } = await helpers.loadFixture(deployPoolFixture);
-
-    const { rm, assetOracle } = await addRiskModuleWithOracles(pool, premiumsAccount, 18, undefined);
-
-    expect(await rm.referenceOracle()).to.equal(hre.ethers.constants.AddressZero);
-
-    await addRound(assetOracle, _E("0.0005")); // 1 ETH = 2000 USD
-
-    expect(await rm.getCurrentPrice()).to.equal(_E("0.0005"));
-  });
-
   it("Should calculate policy premium and loss for single asset with no reference", async () => {
     const { pool, premiumsAccount, accessManager } = await helpers.loadFixture(deployPoolFixture);
 
-    const { rm, assetOracle } = await addRiskModuleWithOracles(pool, premiumsAccount, 18);
+    const { rm, oracle } = await addRiskModuleWithOracles(pool, premiumsAccount);
 
-    await addRound(assetOracle, _E("0.00125")); // 1 ETH = 800 USDC
+    await oracle.setPrice(_E("0.00125")); // 1 ETH = 800 USDC
 
-    const start = await blockchainNow(owner);
+    const start = await helpers.time.latest();
 
-    const [price0, lossProb0] = await rm.pricePolicy(_E("0.001"), true, _A(1000), start + HOUR * 2);
-    expect(price0).to.equal(0);
-    expect(lossProb0).to.equal(0);
+    const [premium0, price0] = await rm.pricePolicy(_E("0.001"), true, _A(1000), start + HOUR * 2);
+    expect(premium0).to.equal(0);
+    expect(price0.lossProb).to.equal(0);
 
     await grantComponentRole(hre, accessManager, rm, "PRICER_ROLE", owner.address);
 
     const priceSlots = await rm.PRICE_SLOTS();
 
     const cdf = new Array(priceSlots);
-    for (let i = 0; i < priceSlots; i++) cdf[i] = _W(i / 100);
-    cdf[priceSlots - 1] = _W("0.5");
+    for (let i = 0; i < priceSlots; i++) {
+      cdf[i] = [_W(i / 100), 0, _W(1)];
+    }
+    cdf[26] = [_W("0.04"), _W("0.3"), _W("0.5")];
+    cdf[priceSlots - 1] = [_W("0.5"), _W(0), _W(1)];
     await rm.connect(owner).setCDF(2, cdf);
 
     // With a variation of 0.4% we have the probability of the first slot
-    let [premium, lossProb] = await rm.pricePolicy(_E("0.00124502"), true, _A(1000), start + HOUR * 2);
-    expect(lossProb).to.equal(_W(0));
+    let [premium, pricing] = await rm.pricePolicy(_E("0.00124502"), true, _A(1000), start + HOUR * 2);
+    expect(pricing.lossProb).to.equal(_W(0));
```

Comment:
At least some of the tests should assert on the collRatios as well, just in case we decide to change the internal implementation and break something.